phoenix-hbase-compat-2.4.0/src/main/java/org/apache/phoenix/compat/hbase/CompatStoreFileReader.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.compat.hbase;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.fs.HFileSystem;
+import org.apache.hadoop.hbase.io.FSDataInputStreamWrapper;
+import org.apache.hadoop.hbase.io.hfile.CacheConfig;
+import org.apache.hadoop.hbase.io.hfile.HFileInfo;
+import org.apache.hadoop.hbase.io.hfile.ReaderContext;
+import org.apache.hadoop.hbase.io.hfile.ReaderContext.ReaderType;
+import org.apache.hadoop.hbase.regionserver.StoreFileReader;
+
+public class CompatStoreFileReader extends StoreFileReader {
+
+    public CompatStoreFileReader(final FileSystem fs, final Path p,
+            final FSDataInputStreamWrapper in, long size, final CacheConfig cacheConf,
+            boolean primaryReplicaStoreFile, AtomicInteger refCount, final Configuration conf)
+            throws IOException {
+        super(new ReaderContext(p, in, size, new HFileSystem(fs), primaryReplicaStoreFile,
+                ReaderType.STREAM),
```

Comment:
I know that HBase has configuration which controls whether or not stream reads happen (compared to pread). I'm not sure if that configuration happens before the StoreFileReader or after. Should we be checking the configuration to determine the ReaderType to use?

Reply:
This was one of the issues that I spent quite a bit of time on when adding 2.3 support (and this is the same as the 2.3 version)
The ReaderType set here will get overwritten in 

org.apache.hadoop.hbase.regionserver.LocalIndexStoreFileScanner.LocalIndexStoreFileScanner(IndexHalfStoreFileReader, boolean, boolean, boolean, long, long, boolean)

so it doesn't matter.